exercises/alice_russell/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ShapeSorter.h"
+
+bool ShapeSorter::compare_area(Shape* &s1, Shape* &s2)
+{
+	return s1->find_area() > s2->find_area();
+}
+
+bool ShapeSorter::compare_perimeter(Shape* &s1, Shape* &s2)
+{
+	return s1->find_perimeter() > s2->find_perimeter();
+}
+
+void ShapeSorter::print_type(std::ostream &out, std::vector<Shape*> &all, std::string &type) const
```

Comment:
`const` is also important in function definitions when using `&` to ensure that a function does not change that parameter. For example, `type` would be written `const std::string &type` as it shouldn't change that. `out` however has to be changed by that function so it is not `const` and can't be.

Reply:
There are similar cases in all of the following methods.